libraries/conf_helpers.rb

```diff
@@ -2,9 +2,13 @@ module CernerSplunk
   # Helper methods for reading and evaluating Splunk config
   module ConfHelpers
     def self.evaluate_config(current_config, desired_config)
-      return desired_config.call(current_config) if desired_config.is_a? Proc
+      desired_config = desired_config.call(current_config) if desired_config.is_a? Proc
       desired_config.map do |section, props|
-        props.is_a?(Proc) ? props.call(section, current_config[section] || {}) : [section, props]
+        mapout = props.is_a?(Proc) ? props.call(section, current_config[section] || {}) : [section, props]
+        mapout[1] = mapout[1].map do |key, value|
+          value.is_a?(Proc) ? value.call(key, (current_config[section] || {})[key]) : [key, value]
```

Comment:
Do we want to talk about the proc arguments? why push current_config as an argument? Also should each layer include the keys from above? Should the top be app and stanza name? What are the set of procs you forsee at each and why would they use these keys?

Also at each layer, do we want to bug out if it's a String instead of a map?